microcosm_postgres/temporary/factories.py

```diff
@@ -0,0 +1,57 @@
+"""
+Temporary table support.
+
+"""
+from types import MethodType
+
+from microcosm_postgres.temporary.methods import (
+    insert_many,
+    select_from,
+    upsert_into,
+)
+
+
+def create_temporary_table(from_table, name=None, on_commit=None):
+    """
+    Create a new temporary table from another table.
+
+    """
+    from_table = from_table.__table__ if hasattr(from_table, "__table__") else from_table
+
+    metadata = from_table.metadata
+    name = name or f"temporary_{from_table.name}"
+
+    # copy the origin table into the metadata with the new name.
+    if name in metadata.tables:
+        temporary_table = metadata.tables[name]
+    else:
+        temporary_table = from_table.tometadata(
+            metadata=metadata,
+            name=name,
+        )
+
+    # change create clause to: CREATE TEMPORARY TABLE
+    temporary_table._prefixes = list(from_table._prefixes)
+    temporary_table._prefixes.append("TEMPORARY")
```

Comment:
Would sqlalchemy's [`prefix_with`](http://docs.sqlalchemy.org/en/latest/core/selectable.html#sqlalchemy.sql.expression.HasPrefixes.prefix_with)

Reply:
It would work on a specific `insert` statement, but not here. I considered doing that, but there's no such modifier for injecting the SQLAlchemy dialect options, so I decided it was better to do all this mucking in one place.